VocaDbModel/Database/Queries/VenueQueries.cs

```diff
@@ -0,0 +1,288 @@
+using System;
+using System.Linq;
+using VocaDb.Model.Database.Repositories;
+using VocaDb.Model.DataContracts.Venues;
+using VocaDb.Model.Domain;
+using VocaDb.Model.Domain.Activityfeed;
+using VocaDb.Model.Domain.ExtLinks;
+using VocaDb.Model.Domain.Security;
+using VocaDb.Model.Domain.Users;
+using VocaDb.Model.Domain.Venues;
+using VocaDb.Model.Helpers;
+using VocaDb.Model.Service;
+using VocaDb.Model.Service.QueryableExtenders;
+using VocaDb.Model.Service.Search.Venues;
+using VocaDb.Model.Service.Translations;
+
+namespace VocaDb.Model.Database.Queries {
+
+	public class VenueQueries : QueriesBase<IVenueRepository, Venue> {
+
+		private readonly IEntryLinkFactory entryLinkFactory;
+		private readonly IEnumTranslations enumTranslations;
+
+		public VenueQueries(IVenueRepository venueRepository, IEntryLinkFactory entryLinkFactory, IUserPermissionContext permissionContext, IEnumTranslations enumTranslations)
+			: base(venueRepository, permissionContext) {
+
+			this.entryLinkFactory = entryLinkFactory;
+			this.enumTranslations = enumTranslations;
+
+		}
+		
+		private ArchivedVenueVersion Archive(IDatabaseContext<Venue> ctx, Venue venue, VenueDiff diff, EntryEditEvent reason, string notes) {
+
+			var agentLoginData = ctx.OfType<User>().CreateAgentLoginData(permissionContext);
+			var archived = ArchivedVenueVersion.Create(venue, diff, agentLoginData, reason, notes);
+			ctx.Save(archived);
+			return archived;
+
+		}
+
+		public (bool created, int reportId) CreateReport(int venueId, VenueReportType reportType, string hostname, string notes, int? versionNumber) {
+
+			ParamIs.NotNull(() => hostname);
+			ParamIs.NotNull(() => notes);
+
+			return HandleTransaction(ctx => {
+				return new Model.Service.Queries.EntryReportQueries().CreateReport(ctx, PermissionContext,
+					entryLinkFactory,
+					(song, reporter, notesTruncated) => new VenueReport(song, reportType, reporter, hostname, notesTruncated, versionNumber),
+					() => reportType != VenueReportType.Other ? enumTranslations.Translation(reportType) : null,
+					venueId, reportType, hostname, notes);
+			});
+
+		}
+		
+		private void CreateTrashedEntry(IDatabaseContext ctx, Venue venue, string notes) {

```

Comment:
So you created all this already huh. I was thinking we wouldn't need this for venues, but I guess it's fine since it's already there.